meson.build

```diff
@@ -0,0 +1,121 @@
+## Experimetal meson build system for dinit
+project(
+    'dinit', 
+    'cpp', 
+    version : run_command(
+    [find_program('sh'), 'build/general_variables_for_meson.sh', 'version'],
+    check : true
+    ).stdout(),
+    license : 'Apache',
+    meson_version : '>= 0.47.0',
+    default_options : ['cpp_std = c++11',
+                       'b_lto = true',
+                       'prefix=/usr'])
+
+## General Defines
+compiler = meson.get_compiler('cpp')
+mconfig_data = configuration_data()
+version = meson.project_version()
+month = run_command('build/general_variables_for_meson.sh', 'month', check : true).stdout()
+year = run_command('build/general_variables_for_meson.sh', 'year', check : true).stdout()
+build_shutdown = get_option('build_shutdown')
+shutdown_prefix = get_option('shutdown_prefix')
+dinit_control_socket_path = get_option('dinit_control_socket_path')
+use_new_abi = get_option('use_new_abi')
+unit_tests = get_option('unit_tests')
+igr_tests = get_option('igr_tests')
+man_pages = get_option('man_pages')
+support_cgroups = get_option('support_cgroups')
+use_utmpx = get_option('use_utmpx')
+# We have custom sbindir for install programs (/sbin instead of /usr/sbin/ by default)
+# By default: Prefix = /usr
+# By default: Sbin = /sbin
+# By default: you find dinit on /sbin/dinit
+# Note: Dinit dont follow Meson's default sbindir; Use dinit_sbindir option instead!
+prefix = get_option('prefix')
+sbindir = get_option('dinit_sbindir')
+mandir = get_option('mandir')
+
+## If Meson's default sbindir is modifed; Warn users about it:
+if get_option('sbindir') != 'sbin'
+  warning('Seems like Default "sbindir" is modified. Dinit dont follow Default "sbindir" option. Please use "dinit_sbindir" option instead!')
+endif
+
+## Use -D_GLIBCXX_USE_CXX11_ABI?
+if use_new_abi.auto()
+  if compiler.get_id() == 'gcc'
```

Comment:
this is bad, gcc != libstdc++

Reply:
i think the build system should just not mess with this define at all, it's not portable, impossible to check, and you can just pass it via `CXXFLAGS` from the environment if you care

Reply:
The problem is the wrong ABI will cause a crash.
The best case is to have code that doesn't compile/link if user choose the wrong ABI. This is the best case and is also supported in meson.

Reply:
see https://github.com/davmac314/dinit/pull/110#issuecomment-1250012202

Reply:
there will be no crash regardless of the ABI used, because libstdc++ supports both and dinit does not link against any third party libraries where this would matter

this is not something for the build system to concern itself with, especially with extremely fragile autodetection like this, just stick with defaults and if somebody wants something else they can pass it via CXXFLAGS

Reply:
but `BUILD` file said:
> Since Dinit is affected by this bug, the unfortunate possibility exists to break Dinit by
upgrading GCC. If you have libstdc++ corresponding to GCC 5.x or 6.x, you *must* build with
the old ABI, but Dinit will be broken if you upgrade to GCC 7. If you have libstdc++ from
GCC 7, you *must* build with the new ABI. If the wrong ABI is used, Dinit may still run
successfully but any attempt to load a non-existing service, for example, will cause Dinit
to crash.

Reply:
that appears related to some ancient gcc/libstdc++ bug, but considering both gcc 5 and 6 are long EOL, i see no reason to care here (even if somebody did build with gcc5/6, they will not do so with meson, so you are introducing pointless clutter)

Reply:
I agree.